packages/core/src/connectors/aliyun-dm/index.ts

```diff
@@ -16,3 +23,43 @@ export const metadata: ConnectorMetadata = {
       '邮件推送（DirectMail）是款简单高效的电子邮件群发服务，构建在阿里云基础之上，帮您快速、精准地实现事务邮件、通知邮件和批量邮件的发送。',
   },
 };
+
+interface Template {
+  type: keyof EmailMessageTypes;
+  subject: string;
+  content: string; // With variable {{code}}, support HTML
+}
+
+export interface AliyunDmConfig {
+  accessKeyId: string;
+  accessKeySecret: string;
+  accountName: string;
+  fromAlias?: string;
+  templates: Template[];
+}
+
+export const sendMessage: EmailSendMessageFunction = async (address, type, data) => {
+  const config: AliyunDmConfig = await getConnectorConfig<AliyunDmConfig>(
+    metadata.id,
+    metadata.type
+  );
+  const template = config.templates.find((template) => template.type === type);
+
+  if (!template) {
+    throw new ConnectorError(`Can not find template for type: ${type}`);
+  }
+
+  return sendSingleMail(
+    {
+      AccessKeyId: config.accessKeyId,
+      AccountName: config.accountName,
+      ReplyToAddress: 'false',
+      AddressType: '1',
+      ToAddress: address,
+      FromAlias: config.fromAlias,
+      Subject: template.subject,
+      HtmlBody: template.content.replaceAll('{{code}}', data.code),
```

Comment:
node 16 supports `replaceAll` already?

Reply:
Can not find document for now, but YES, tested.